libbpf-rs/src/perf_buffer.rs

```diff
@@ -142,6 +161,49 @@ impl<'a, 'b> PerfBufferBuilder<'a, 'b> {
         }
     }
 
+    unsafe extern "C" fn call_event_cb(
+        ctx: *mut c_void,
+        cpu: i32,
+        hdr: *mut libbpf_sys::perf_event_header,
+    ) -> i32 {
+        #[repr(C)]
+        #[derive(Debug, Default, Copy, Clone)]
+        struct perf_sample_raw {
+            pub header: libbpf_sys::perf_event_header,
+            pub size: u32,
+            pub data: [u8; 0],
+        }
+
+        #[repr(C)]
+        #[derive(Debug, Default, Copy, Clone)]
+        struct perf_sample_lost {
+            pub header: libbpf_sys::perf_event_header,
+            pub id: u64,
+            pub lost: u64,
+            pub sample_id: u64,
+        }
+
+        match (*hdr).type_ {
+            libbpf_sys::PERF_RECORD_SAMPLE => {
+                let s = hdr as *mut perf_sample_raw;
+                let event_sample = &mut *s;
+                Self::call_sample_cb(
+                    ctx,
+                    cpu,
+                    std::ptr::addr_of_mut!(event_sample.data) as *mut c_void,
```

Comment:
Can you please use `ptr::` for consistency? Here and for all other uses you introduce.

Reply:
Done